### DISCUSSION TITLE
How about an option to cycle through shader types of models without the need to do so on the .cfg file, like a vcommand of sorts?

### DISCUSSION THREAD

**Discussion by SniperGoth** [redeclipse/base#1145] [Ideas and Feature Requests]
This is something that i'd find quite useful, to be able to change the shader on the go, with no need to mess the .cfg file every time.
A example:
So let's say i have a model with envmap (model1), and i want to disable it for that map.
In this case, i'd need to create a (model1-noenv), which would be a duplicate of the original.
Then mess with the cfg a bit, load it up in-game to see the results.
Or you have a model with lights, and want to disable that.
After all, we can already be able to change general colors of them, like vcolor's.
Would be nice to have the option of changing things on the go.

**Comment by qreeves** [comment]
You'd never be able to represent all the options in a way that'd make that work properly. The proper way to go about it is:

- In your base model folder ("model1"), create a subfolder ("noenv", path is now "model1/noenv"). Copy the ".cfg" and add "../" to the beginning of file paths. The only thing you're duplicating now is the config file (which is the point).
- Load the model, then use `/clearmodel "model1/noenv"` to reload the model.

See the weapon model folders for the best example of this in action.

**Comment by SniperGoth** [comment]
I guess that's how i should do it then.

**Comment by cloudziq** [comment]
but we still need an ability to use another model for mapvariant (like one with glow for night, second without a glow for a day)
